### PR TITLE
no string-to-array

### PR DESCRIPTION
**Pull request by snadrus**
Stop using broken YB function string_to_array()

### CODE REVIEW

File: tasks/window/compute_task.go [hunk 1]
```diff
@@ -289,14 +288,14 @@ func (t *WdPostTask) CanAccept(ids []harmonytask.TaskID, te *harmonytask.TaskEng
...
+		`SELECT 
 			task_id,
 			sp_id,
 			proving_period_start,
 			deadline_index,
 			partition_index
-	from wdpost_partition_tasks 
-	where task_id IN (SELECT unnest(string_to_array($1, ','))::bigint)`, strings.Join(lo.Map(ids, entToStr[harmonytask.TaskID]), ","))
+	FROM wdpost_partition_tasks 
+	WHERE task_id IN ($1)`, lo.Map(ids, func(t harmonytask.TaskID, _ int) int { return int(t) }))
```
Comment (LexLuthr) [inline tasks/window/compute_task.go:298]:
```suggestion
	WHERE task_id = ANY($1)`, lo.Map(ids, func(t harmonytask.TaskID, _ int) int { return int(t) }))
```

**Review by LexLuthr — CHANGES_REQUESTED** [review]
Postgres syntax is incorrect for lookup in array

File: tasks/window/compute_task.go [hunk 2]
```diff
@@ -289,14 +288,14 @@ func (t *WdPostTask) CanAccept(ids []harmonytask.TaskID, te *harmonytask.TaskEng
...
+		`SELECT 
 			task_id,
 			sp_id,
 			proving_period_start,
 			deadline_index,
 			partition_index
-	from wdpost_partition_tasks 
-	where task_id IN (SELECT unnest(string_to_array($1, ','))::bigint)`, strings.Join(lo.Map(ids, entToStr[harmonytask.TaskID]), ","))
+	FROM wdpost_partition_tasks 
+	WHERE task_id IN ANY($1)`, lo.Map(ids, func(t harmonytask.TaskID, _ int) int { return int(t) }))
```
Comment (LexLuthr) [inline tasks/window/compute_task.go:298]:
```suggestion
	WHERE task_id IN ANY($1)`, lo.Map(ids, func(t harmonytask.TaskID, _ int) int64 { return int64(t) }))
```

**Review by LexLuthr — CHANGES_REQUESTED** [review]
task_id is a bigint type in DB so we need to match that in Golang

**Comment by snadrus** [comment]
That's not a problem at all:
https://www.postgresql.org/docs/current/functions-comparison.html

**Review by LexLuthr — APPROVED** [review]

**Pull request merged**